tests/jenkins/cloud-native-dev-with-spring-boot.groovy

```diff
@@ -0,0 +1,308 @@
+                        echo ex.toString()
+                        throw ex
+                    }
+                }
+            }
+        }
+
+        stage('Confirm before retiring') {
+            when {
+                expression {
+                    return params.confirm_before_delete
+                }
+            }
+            steps {
+                input "Continue ?"
+            }
+        }
+        stage('Retire service from CF') {
+            environment {
+                uri = "${cf_uri}"
+                credentials = credentials("${opentlc_creds}")
+                admin_credentials = credentials("${opentlc_admin_creds}")
+                DEBUG = 'true'
+            }
+            /* This step uses the delete_svc_guid.sh script to retire
+             the service from CloudForms */
+            steps {
+                git 'https://github.com/redhat-gpte-devopsautomation/cloudforms-oob'
+
+                sh "./opentlc/delete_svc_guid.sh '${guid}'"
+            }
+            post {
+                failure {
+                    withCredentials([usernameColonPassword(credentialsId: imap_creds, variable: 'credentials')]) {
+                        mail(
+                            subject: "${env.JOB_NAME} (${env.BUILD_NUMBER}) failed retiring for GUID=${guid}",
+                            body: "It appears that ${env.BUILD_URL} is failing, somebody should do something about that.\nMake sure GUID ${guid} is destroyed.",
+                            to: "${notification_email}",
+                            replyTo: "${notification_email}",
+                            from: credentials.split(':')[0]
+                        )
+                    }
+                    withCredentials([string(credentialsId: rocketchat_hook, variable: 'HOOK_URL')]) {
+                        sh(
+                            """
+                            curl -H 'Content-Type: application/json' \
+                            -X POST '${HOOK_URL}' \
+                            -d '{\"username\": \"jenkins\", \"icon_url\": \"https://dev-sfo01.opentlc.com/static/81c91982/images/headshot.png\", \"text\": \"@here :rage: ${env.JOB_NAME} (${env.BUILD_NUMBER}) failed retiring ${guid}.\"}'\
+                            """.trim()
+                        )
+                    }
+                }
+            }
+        }
+        stage('Wait for deletion email') {
+            steps {
+                git url: 'https://github.com/redhat-cop/agnosticd',
+                    branch: 'development'
+
+                withCredentials([usernameColonPassword(credentialsId: imap_creds, variable: 'credentials')]) {
+                    sh """./tests/jenkins/downstream/poll_email.py \
+                        --guid ${guid} \
+                        --timeout 20 \
+                        --server '${imap_server}' \
+                        --filter 'has been deleted'"""
+                }
+            }
+        }
+    }
+
```

Comment:
You never check that the environment is actually gone

A better test would be:

- once the COMPLETED email is received, use the information from the email content to connect to the environment. For a shared catalog item (workload deployed to a shared OCP environment) we can test if the project is created.
- after the `has been deleted` email is received, make sure that same connection doesn't work, or at least the the project is deleted on the shared cluster.

Reply:
This will look into next week probably on Tuesday.